File: iosApp/iosApp/UI/Screens/HistoryScreen.swift
import SwiftUI

// MARK: - History Screen
struct HistoryScreen: View {
    let historyEntries: [HistoryEntry]

    @Environment(\.colorScheme) var colorScheme
    @Environment(\.dismiss) private var dismiss

    private var backgroundColor: Color {
        colorScheme == .dark ? ErgoGuardColors.Dark.background : ErgoGuardColors.background
    }

    private var textColor: Color {
        colorScheme == .dark ? ErgoGuardColors.Dark.onBackground : ErgoGuardColors.onBackground
    }

    private var surfaceColor: Color {
        colorScheme == .dark ? ErgoGuardColors.Dark.surfaceVariant : ErgoGuardColors.surfaceVariant
    }

    var body: some View {
        ZStack {
            backgroundColor
                .ignoresSafeArea()

            if historyEntries.isEmpty {
                emptyState
            } else {
                historyList
            }
        }
        .navigationTitle("history_title")
        .navigationBarTitleDisplayMode(.large)
        .toolbarBackground(.visible, for: .navigationBar)
    }

    // MARK: - Empty State
    private var emptyState: some View {
        VStack(spacing: 16) {
            Image(systemName: "chart.bar.doc.horizontal")
                .font(.system(size: 64))
                .foregroundColor(textColor.opacity(0.4))

            Text("history_empty_title")
                .font(ErgoGuardTypography.headlineSmall)
                .foregroundColor(textColor.opacity(0.6))

            Text("history_empty_desc")
                .font(ErgoGuardTypography.bodyMedium)
                .foregroundColor(textColor.opacity(0.4))
        }
    }

    // MARK: - History List
    private var historyList: some View {
        ScrollView {
            LazyVStack(spacing: 12) {
                ForEach(historyEntries.sorted(by: { $0.timestamp > $1.timestamp })) { entry in
                    HistoryCard(entry: entry)
                }
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 16)
        }
    }
}

// MARK: - History Card
struct HistoryCard: View {
    let entry: HistoryEntry

    @Environment(\.colorScheme) var colorScheme

    private var textColor: Color {
        colorScheme == .dark ? ErgoGuardColors.Dark.onSurfaceVariant : ErgoGuardColors.onSurfaceVariant
    }

    private var surfaceColor: Color {
        colorScheme == .dark ? ErgoGuardColors.Dark.surfaceVariant : ErgoGuardColors.surfaceVariant
    }

    private let dateFormatter: DateFormatter = {
        let formatter = DateFormatter()
        formatter.dateFormat = "dd/MM/yyyy HH:mm"
        formatter.locale = Locale(identifier: "vi_VN")
        return formatter
    }()

    var body: some View {
        HStack(spacing: 16) {
            // Percentage badge
            ZStack {
                RoundedRectangle(cornerRadius: 12)
                    .fill(entry.level.color.opacity(0.15))
                    .frame(width: 64, height: 64)

                Text("\(Int(entry.percentage))%")
                    .font(ErgoGuardTypography.titleLarge)
                    .fontWeight(.bold)
                    .foregroundColor(entry.level.color)
            }

            // Details
            VStack(alignment: .leading, spacing: 4) {
                Text(entry.level.label)
                    .font(ErgoGuardTypography.titleMedium)
                    .fontWeight(.semibold)
                    .foregroundColor(entry.level.color)

                Text("pressure_label \(Int(entry.neckLoadKg))")
                    .font(ErgoGuardTypography.bodyMedium)
                    .foregroundColor(textColor)

                Text(dateFormatter.string(from: entry.timestamp))
                    .font(ErgoGuardTypography.bodySmall)
                    .foregroundColor(textColor.opacity(0.7))
            }

            Spacer()
        }
        .padding(16)
        .background(
            RoundedRectangle(cornerRadius: 16)
                .fill(surfaceColor.opacity(colorScheme == .dark ? 0.5 : 0.7))
        )
        .overlay(
            RoundedRectangle(cornerRadius: 16)
                .stroke(surfaceColor.opacity(0.3), lineWidth: 1)
        )
    }
}

// MARK: - Preview
#if DEBUG
struct HistoryScreen_Previews: PreviewProvider {
    static var previews: some View {
        Group {
            NavigationStack {
                HistoryScreen(
                    historyEntries: [
                        HistoryEntry(
                            timestamp: Date(),
                            percentage: 15,
                            level: .normal,
                            neckLoadKg: 5
                        ),
                        HistoryEntry(
                            timestamp: Date().addingTimeInterval(-86400),
                            percentage: 45,
                            level: .warning,
                            neckLoadKg: 12
                        ),
                        HistoryEntry(
                            timestamp: Date().addingTimeInterval(-172800),
                            percentage: 85,
                            level: .danger,
                            neckLoadKg: 27
                        )
                    ]
                )
            }
            .preferredColorScheme(.light)

            NavigationStack {
                HistoryScreen(historyEntries: [])
            }
            .preferredColorScheme(.dark)
        }
    }
}
#endif

File: composeApp/src/androidMain/kotlin/com/example/ergoguard/ui/HistoryScreen.kt
package com.example.ergoguard.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.ergoguard.data.HistoryEntry
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.ArrowBack
import java.text.SimpleDateFormat
import java.util.*

import android.app.Activity
import androidx.compose.ui.platform.LocalView
import androidx.core.view.WindowCompat
import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.ui.graphics.toArgb
import androidx.compose.ui.res.stringResource
import com.example.ergoguard.R

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun HistoryScreen(
    historyEntries: List<HistoryEntry>,
    onBackClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    val view = LocalView.current
    val isDarkTheme = isSystemInDarkTheme()
    val backgroundColor = MaterialTheme.colorScheme.background
    if (!view.isInEditMode) {
        SideEffect {
            val window = (view.context as Activity).window
            window.statusBarColor = backgroundColor.toArgb()
            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = !isDarkTheme
        }
    }

    Scaffold(
        modifier = modifier.fillMaxSize(),
        containerColor = MaterialTheme.colorScheme.background,
        topBar = {
            TopAppBar(
                title = { Text(stringResource(R.string.history_title), style = MaterialTheme.typography.titleLarge) },
                navigationIcon = {
                    IconButton(onClick = onBackClick) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                            contentDescription = stringResource(R.string.cd_back),
                            tint = MaterialTheme.colorScheme.onSurface
                        )
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.surface,
                    titleContentColor = MaterialTheme.colorScheme.onSurface
                )
            )
        }
    ) { paddingValues ->
        if (historyEntries.isEmpty()) {
            // Empty state
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(paddingValues),
                contentAlignment = Alignment.Center
            ) {
                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    Text("📊", fontSize = 64.sp)
                    Spacer(modifier = Modifier.height(16.dp))
                    Text(
                        stringResource(R.string.history_empty_title),
                        style = MaterialTheme.typography.titleMedium,
                        color = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.6f)
                    )
                    Text(
                        stringResource(R.string.history_empty_desc),
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.4f)
                    )
                }
            }
        } else {
            LazyColumn(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(paddingValues)
                    .padding(horizontal = 16.dp),
                verticalArrangement = Arrangement.spacedBy(12.dp),
                contentPadding = PaddingValues(vertical = 16.dp)
            ) {
                items(historyEntries.sortedByDescending { it.timestamp }) { entry ->
                    HistoryCard(entry = entry)
                }
            }
        }
    }
}

@Composable
private fun HistoryCard(entry: HistoryEntry) {
    val levelColor = when (entry.level) {
        "NORMAL" -> MaterialTheme.colorScheme.tertiary // Emerald/Green from M3 palette
        "WARNING" -> Color(0xFFB98200) // Warning Orange
        "DANGER" -> MaterialTheme.colorScheme.error
        else -> MaterialTheme.colorScheme.outline
    }

    val statusText = when (entry.level) {
        "NORMAL" -> stringResource(R.string.status_safe)
        "WARNING" -> stringResource(R.string.status_warning_mild)
        "DANGER" -> stringResource(R.string.status_warning_severe)
        else -> entry.level
    }

    val dateFormat = remember { SimpleDateFormat("dd/MM/yyyy HH:mm", Locale.getDefault()) }
    val formattedDate = dateFormat.format(Date(entry.timestamp))

    Card(
        modifier = Modifier.fillMaxWidth(),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surfaceVariant,
            contentColor = MaterialTheme.colorScheme.onSurfaceVariant
        )
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Percentage badge
            Box(
                modifier = Modifier
                    .size(64.dp)
                    .background(levelColor.copy(alpha = 0.15f), RoundedCornerShape(12.dp)),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = "${entry.percentage.toInt()}%",
                    style = MaterialTheme.typography.titleLarge,
                    color = levelColor,
                    fontWeight = FontWeight.Bold
                )
            }

            Spacer(modifier = Modifier.width(16.dp))

            // Details
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = statusText,
                    style = MaterialTheme.typography.titleMedium,
                    color = levelColor,
                    fontWeight = FontWeight.SemiBold
                )
                Text(
                    text = "${stringResource(R.string.history_load_prefix)} ${entry.neckLoadKg.toInt()} kg",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
                Text(
                    text = formattedDate,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.7f)
                )
            }
        }
    }
}

@androidx.compose.ui.tooling.preview.Preview(showBackground = true)
@Composable
private fun HistoryScreenPreview() {
    com.example.ergoguard.ui.theme.ErgoGuardTheme {
        HistoryScreen(
            historyEntries = listOf(
                HistoryEntry(
                    timestamp = System.currentTimeMillis(),
                    percentage = 15.0,
                    level = "NORMAL",
                    neckLoadKg = 5.0
                ),
                HistoryEntry(
                    timestamp = System.currentTimeMillis() - 86400000,
                    percentage = 45.0,
                    level = "WARNING",
                    neckLoadKg = 12.0
                ),
                HistoryEntry(
                    timestamp = System.currentTimeMillis() - 172800000,
                    percentage = 85.0,
                    level = "DANGER",
                    neckLoadKg = 27.0
                )
            ),
            onBackClick = {}
        )
    }
}
